expression/builtin_time.go

```diff
@@ -745,3 +745,24 @@ func parseDayInterval(sc *variable.StatementContext, value types.Datum) (int64,
 	}
 	return value.ToInt64(sc)
 }
+
+// https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_timestampdiff
+func builtinTimestampDiff(args []types.Datum, ctx context.Context) (d types.Datum, err error) {
+	sc := ctx.GetSessionVars().StmtCtx
+	t1, err := convertDatumToTime(sc, args[1])
+	if err != nil {
+		return
```

Comment:
May be it depends on sql_mode and statement type? Please try to insert timestampdiff(MONTH, 'asdlfjasdf', '2017-01-01')  into some table.